src/coreclr/jit/lsrabuild.cpp

```diff
@@ -851,15 +851,23 @@ regMaskTP LinearScan::getKillSetForCall(GenTreeCall* call)
     if (!compiler->compFloatingPointUsed)
     {
 #if defined(TARGET_XARCH)
-        killMask &= ~(RBM_FLT_CALLEE_TRASH | RBM_MSK_CALLEE_TRASH);
+
+#ifdef TARGET_AMD64
+        killMask.RemoveRegsetForType(RBM_FLT_CALLEE_TRASH.getLow(), FloatRegisterType);
+        killMask.RemoveRegsetForType(RBM_MSK_CALLEE_TRASH.getLow(), MaskRegisterType);
 #else
-        killMask &= ~RBM_FLT_CALLEE_TRASH;
+        killMask.RemoveRegsetForType(RBM_FLT_CALLEE_TRASH, FloatRegisterType);
+        killMask &= ~RBM_MSK_CALLEE_TRASH;
+#endif // TARGET_AMD64
+
+#else
+        killMask.RemoveRegsetForType(RBM_FLT_CALLEE_TRASH, FloatRegisterType);
```

Comment:
I guess this will change back to what it was before once we change `RBM_*` to be `regMaskTP`?

Reply:
I don't want to block on it, but I guess it would be nice to make the `RBM_*` change first, since it seems a lot of the changes of this PR end up having to deal with the fact that `RBM_*` aren't `regMaskTP`.

Reply:
Sure, there are few places that we have to do, but at this point I think I will go ahead and merge this PR as-is and probably start working on `RBM_` change next.